### ISSUE TITLE
Allow the timestamp field to be customized

### ISSUE THREAD

**Issue by savaki** [uber-go/zap#33]
Currently, the timestamp is defined to be unix nanos within the field, "ts".  I'd like to be able to pump the logs directly into third party services like loggly which have a different naming requirement and format for the timestamp field.

Was thinking of something like this:

https://github.com/savaki/zap

**Comment by akshayjshah** [comment]
@savaki I've never used Loggly before. From https://www.loggly.com/docs/automated-parsing/#json, it looks we want the flexibility to (a) call the field `timestamp`, and (b) encode `time.Time`s using ISO-8601. Is that right?

**Comment by savaki** [comment]
Correct.  I'm using my forked version now to publish logs to loggly.

``` go
func appendTimestamp(data []byte, t time.Time) []byte {
	data = append(data, `,"timestamp":"`...)
	data = t.UTC().AppendFormat(data, "2006-01-02T15:04:05.000Z")
	return append(data, `"`...)
}

func sample() {
	logger = zap.NewJSON(zap.Output(output), zap.Append(appendTimestamp))
}
```

**Comment by akshayjshah** [comment]
Gotcha. I think we can actually provide a little more flexibility than that - it should be easy enough to allow users to configure nearly everything about the final message (time format, key names, etc.).

I'll try to open a PR this weekend. When I do, I'll tag you - let me know what you think.

**Comment by savaki** [comment]
awesome, thanks.

**Comment by akshayjshah** [comment]
Note to self: this should also replace the `StubTime` kludge currently in the Logger interface.

**Comment by garo** [comment]
+1 here, we require our 'ts' field to be ISO-8601 format (ie. "2006-01-02T15:04:05.000Z")

**Comment by akshayjshah** [comment]
Addressed in #115. You can now customize your timestamps with something like this:

```go
logger := zap.New(
  zap.NewJSONEncoder(zap.RFC3339Formatter("@timestamp")),
  zap.ErrorLevel,
  zap.Output(someFile),
)
```

If your preferred time format isn't supported out of the box, it should be easy to plug in your own `TimeFormatter`.
